cassiopeia/data/CassiopeiaTree.py

```diff
@@ -1872,6 +1872,105 @@ def get_distances(
             for _node in (self.leaves if leaves_only else self.nodes)
         }
 
+    def scale_to_unit_length(self) -> None:
+        """
+        Scales the tree to have unit length.
+
+        The longest path from root to leaf will have length 1 after the
+        scaling.
+
+        Raises:
+            CassiopeiaTreeError if the tree has not been initialized.
+        """
+        self.__check_network_initialized()
+
+        current_times = self.get_times().values()
+        max_time = max(current_times)
+        min_time = min(current_times)
+        new_times = {}
+        for node in self.nodes:
+            new_times[node] = self.get_time(node) / (max_time - min_time)
```

Comment:
Isn't this supposed to be `(time - min) / (max - min)`? Or is this functionally the same?

Reply:
Either way, the tree will have unit length after the scaling. However, I like your proposal better because it makes the root have time 0, so I will push this change.

Note that if we were to enforce that throughout the lifetime of the tree the root has time zero, this code can be simplified to `time  / max`. But I don't think we are enforcing this constraint, not it's clear to me that we should.

Reply:
Good call -- I understand now why you were initially just dividing `time / (max - min)`, but yeah I think we should be more specific here and move to `(time - min) / (max-min)`.